### PR TITLE
Prevent a link from being created twice

### PR DESCRIPTION
**Pull request by geoand**
## What is the purpose of this change? What does it change?
See $SUBJECT

## Was the change discussed in an issue?

Fixes: #1201

## How to test changes?
Create a backend component and then a fronted.
The first attempt to link to the backend should work.
The second one will fail.
Note that when ports are needed, links to a different ports are allowed - but not multiple links to the same port.

Also linking from a component to a service is also only allowed once

### CODE REVIEW

**Comment by geoand** [comment]
@jorgemoralespou I added a fix for #1201. I haven't included the prefix part however.
Is that something that we need? Do you think it's useful?

File: pkg/component/component.go [hunk 1]
```diff
@@ -201,6 +201,25 @@ func GetComponentPorts(client *occlient.Client, componentName string, applicatio
 	return ports, nil
 }
 
+// GetComponentLinkedSecretNames provides a slice containing the names of the secrets that are present in envFrom
+func GetComponentLinkedSecretNames(client *occlient.Client, componentName string, applicationName string) (secretNames []string, err error) {
```
[the remark this answers is not in the document]
Comment (geoand) [reply pkg/component/component.go]: Hm.. Are we sure we want to pass that abstraction down? I don't see it being used in the other functions of the file

[the remark this answers is not in the document]
Comment (metacosm) [reply pkg/component/component.go]: There was some discussion about using Context for all function needing similar infos but that never was decided. So feel free to ignore ;)

File: pkg/component/component_test.go [hunk 2]
```diff
@@ -97,6 +97,76 @@ func TestGetComponentPorts(t *testing.T) {
...
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		output  []string
+	}{
+		{
+			name: "Case 1: Invalid/Non-existant component name",
```
[the remark this answers is not in the document]
Comment (metacosm) [reply pkg/component/component_test.go]: I'm actually not sure what the proper term should be… :(

[the remark this answers is not in the document]
Comment (geoand) [reply pkg/component/component_test.go]: Me neither... I think they are both wrong, but `unexisting` sounds better to me :P

File: pkg/component/component.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (geoand) [reply pkg/component/component.go]: :+1:

**Comment by girishramnani** [comment]
@geoand e2e test for this would be really great

**Comment by geoand** [comment]
> @geoand e2e test for this would be really great

Makes sense, I'll add that

**Comment by geoand** [comment]
Now I will just wait for tests to pass :)

**Pull request merged**